flow/core/kernel/base_energy.py

```diff
@@ -0,0 +1,23 @@
+from abc import ABCMeta, abstractmethod
+
+
+class BaseEnergyModel(metaclass=ABCMeta):
+    """Base energy model class.
+
+    Calculate the instantaneous power consumption of a vehicle in
+    the network.  It returns the power in Watts regardless of the 
+    vehicle type: whether EV or Combustion Engine, Toyota Prius or Tacoma
+    or non-Toyota vehicles. Non-Toyota vehicles are set by default 
+    to be an averaged-size vehicle.
+    """
+
+    def __init__(self, kernel):
+        self.k.env.vehicle = vehicle
```

Comment:
we'll need to figure out what goes here specifically based on how these classes will be instantiated. follow up on Eugene's suggestions, and let's talk to Sulaiman/Yashar on this.

i think `filename` should be an input here since the Toyota ones need it, so you can just add `filename=None` here.

also, for the vehicle, you are currently assigning `vehicle` to `self.k.env.vehicle`, but there are a couple problems with this.
1. vehicle is an unknown object, unless you add it to the call signature
2. you can't assign anything to `self.k.env.vehicle` until it knows what `self.k.env` is. similarly, you can't assign anything to `self.k.env` until it knows what `self.k` is.

so i think what you'll want to do is something like:
```
def __init__(self, kernel, vehicle, filename=None):
    self.kernel = kernel # i suspect you don't need this, but i'm not sure yet
    self.vehicle = vehicle
```

Reply:
you want to call methods from other sub-kernels (vehicle). to be consistent, you can use "master_kernel" similar to what we have in the other sub-kernel base classes.